superset/charts/api.py

```diff
@@ -450,7 +450,7 @@ def data(self) -> Response:
         except KeyError:
             return self.response_400(message="Request is incorrect")
         try:
-            security_manager.assert_query_context_permission(query_context)
+            query_context.raise_for_access()
```

Comment:
not changed in this PR, but this try/catch seems kinda weird, I would've expected this api to be wrapped in the `handle_api_exception` decorator that i believe automatically generates the proper json error response from an exception

Reply:
@etr2460 I think this merely highlights the inconsistencies with how backend errors are handled. This is detailed in [SIP-41](https://github.com/apache/incubator-superset/issues/9298).

Reply:
@etr2460 #9529 kind of addresses that, it's a POC for SIP-41, is uses flask error handling